### ISSUE TITLE
loss stucked at 3.0

### ISSUE THREAD

**Issue by ddeellttaa** [mitming/AlignDet#26]
Hello, thanks for the nice work! I try to use AlignDet in yolov8, but when I train the model ,I found the contrastive loss stucked at 3.0 after about 50 epochs.

And I check your logs when pretrain fcos and find out the loss of fcos is stuck as well. I wonder if this is normal.

**Comment by liming-ai** [comment]
> Hello, thanks for the nice work! I try to use AlignDet in yolov8, but when I train the model ,I found the contrastive loss stucked at 3.0 after about 50 epochs.
> 
> And I check your logs when pretrain fcos and find out the loss of fcos is stuck as well. I wonder if this is normal.

Hi @ddeellttaa,

Thank you for your question.

During the pre-training stage, only a small number of model parameters (Neck/Head) will be trained, so it is indeed possible that the loss cannot continue to decrease, and the model may have converged at this time.

However, if it is stuck at a specific value (such as 3.0), it is very abnormal. I suggest you review the code and related training data.

**Comment by ddeellttaa** [comment]
> > Hello, thanks for the nice work! I try to use AlignDet in yolov8, but when I train the model ,I found the contrastive loss stucked at 3.0 after about 50 epochs.
> > And I check your logs when pretrain fcos and find out the loss of fcos is stuck as well. I wonder if this is normal.
> 
> Hi @ddeellttaa,
> 
> Thank you for your question.
> 
> During the pre-training stage, only a small number of model parameters (Neck/Head) will be trained, so it is indeed possible that the loss cannot continue to decrease, and the model may have converged at this time.
> 
> However, if it is stuck at a specific value (such as 3.0), it is very abnormal. I suggest you review the code and related training data.

Thanks for your reply. 

It seems like the Aligndet works very well although the loss stucked at 3.0. I use the model pretrained by Aligndet and finetune it, the map@0.5 increase 10, better then any other model I trained.

Again, thanks for your great work!

**Comment by liming-ai** [comment]
> > > Hello, thanks for the nice work! I try to use AlignDet in yolov8, but when I train the model ,I found the contrastive loss stucked at 3.0 after about 50 epochs.
> > > And I check your logs when pretrain fcos and find out the loss of fcos is stuck as well. I wonder if this is normal.
> > 
> > 
> > Hi @ddeellttaa,
> > Thank you for your question.
> > During the pre-training stage, only a small number of model parameters (Neck/Head) will be trained, so it is indeed possible that the loss cannot continue to decrease, and the model may have converged at this time.
> > However, if it is stuck at a specific value (such as 3.0), it is very abnormal. I suggest you review the code and related training data.
> 
> Thanks for your reply.
> 
> It seems like the Aligndet works very well although the loss stucked at 3.0. I use the model pretrained by Aligndet and finetune it, the map@0.5 increase 10, better then any other model I trained.
> 
> Again, thanks for your great work!

Alright😂

Glad to know AlignDet could be helpful for your work, but the loss here seems strange, maybe you can change the loss weight or temperature of contrastive loss to see whether it would be changed.
